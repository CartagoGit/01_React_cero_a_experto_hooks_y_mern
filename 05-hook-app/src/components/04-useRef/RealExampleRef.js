import React, { useState } from "react";
import "../02-useEffect/effects.css";
import { MultipleCustomHooks } from "../03-MultipleHooks-BreakingBad-API/MultipleCustomHooks";

export const RealExampleRef = () => {
	const [show, setShow] = useState(false);
	return (
		<>
			<h1>RealExampleRef</h1>
			<button onClick={() => setShow(!show)} className='btn btn-primary'>
				Show/Hide
			</button>
			<hr />

			{show && (
				<>
					<MultipleCustomHooks />
					<hr />{" "}
				</>
			)}
		</>
	);
};
