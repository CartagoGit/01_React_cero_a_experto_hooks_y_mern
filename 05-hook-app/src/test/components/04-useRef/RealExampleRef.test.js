import React from "react";
import "@testing-library/jest-dom";
import { shallow, simulate } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe("Pruebas en <RealExampleRef />>", () => {
	let wrapper;

	const fillWrapper = () => {
		wrapper = shallow(<RealExampleRef />);
	};
	beforeEach(() => fillWrapper());

	test("Deberia mostrarse correctamente", () => {
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("MultipleCustomHooks").at(0).exists()).toBe(false);
	});

	test("Deberia mostrar el componente <MultipleCustomHooks />", () => {
		wrapper.find("button").at(0).simulate("click");
		// console.log(wrapper.find("MultipleCustomHooks").at(0).html());
		expect(wrapper.find("MultipleCustomHooks").at(0).exists()).toBe(true);
	});
});
