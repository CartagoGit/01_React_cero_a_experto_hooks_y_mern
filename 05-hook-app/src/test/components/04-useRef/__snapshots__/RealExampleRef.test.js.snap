// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef />> Deberia mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
  <hr />
</Fragment>
`;
